script.js

```diff
@@ -1,12 +1,214 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let allEpisodes;
+let showId;
+let homeButton = document.getElementById("button");
+homeButton.addEventListener("click", () => {
+  searchBar.disabled = false;
+  searchBar.innerHTML = "";
+  makePageForShows(allShows);
+  selectSeries.value = null;
+  selectSeries.selectedIndex = 0;
+});
+
+let allShows = getAllShows().sort(function (a, b) {
+  if (a.name < b.name) {
+    return -1;
+  }
+});
+
+function showAllEpisodes(showId) {
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+    .then((response) => {
+      return response.json();
+    })
+    .then(function (data) {
+      allEpisodes = data;
+      makePageForEpisodes(allEpisodes);
+      createDropDownForEpisodes(allEpisodes);
+    })
+    .catch((err) => console.log(err));
 }
 
+const rootElem = document.getElementById("root");
+const mainElm = document.createElement("div");
+var searchBar = document.getElementById("search");
+var selectSeries = document.getElementById("selectSeries");
+let select = document.getElementById("selectBox");
+const display = document.getElementById("display");
+mainElm.id = "main";
+rootElem.appendChild(mainElm);
+
 function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  mainElm.innerHTML = "";
+  episodeList.forEach((episode) => createCards(episode));
+  display.innerText = `Displaying ${episodeList.length}/${allEpisodes.length} episodes`;
+}
+
+function makePageForShows(allShowsList) {
+  select.innerHTML = "";
+  mainElm.innerHTML = "";
+  allShowsList.forEach((show) => createCardsForShows(show));
+  display.innerText = `Displaying ${allShowsList.length}/${allShows.length} shows`;
+}
+
+function createCards(episode) {
+  var cardElm = document.createElement("div");
+  let titleElm = document.createElement("h2");
+  let imageEl = document.createElement("img");
+  let pElm = document.createElement("p");
+
+  titleElm.textContent =
+    episode.name + creatEpisodeCount(episode.season, episode.number);
+  if (episode.image == null) {
+    imageEl.src =
+      "https://upmaa-pennmuseum.netdna-ssl.com/collections/images/image_not_available_300.jpg";
+  } else {
+    imageEl.src = episode.image.medium;
+  }
+  pElm.innerHTML = episode.summary;
+  mainElm.appendChild(cardElm);
+  cardElm.appendChild(titleElm);
+  cardElm.appendChild(imageEl);
+  cardElm.appendChild(pElm);
+  cardElm.id = "card";
+  titleElm.id = "title";
+}
+function createCardsForShows(show) {
```

Comment:
`createCardsForShows` function is quite long and it's responsible for creating many elements. Writing clean code that is easy to read and understand is really important. One of the main principles is that functions should have a single responsibility. Could you please go back to the pull request #1 and read my comment about it again? Elements in the DOM have their hierarchy, for example your card element is a parent of title, image and summary elements so it should only know about them and not all of the elements. Functions responsible for creating other elements, such as genres paragraph, should be called in the function that creates the footer as it's the parent of that element. I know that this is a lot of words and it may be confusing at the beginning, so if you will be struggling for too long with refactoring this please let me know and I'll be happy to explain it further on the call🙂

`createCards` function could be also broken down into smaller functions in a similar way.